src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -47,19 +47,32 @@ This is a sphere with inner radius 1 and outer radius 2.
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP 
+    θ::TT 
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+    
+    function Ellipsoid{T,CO}(r::TR,φ::TP,θ::TT,origin::CartesianPoint{T}=zero(CartesianPoint{T}),
+        rotation::SMatrix{3,3,T,9}=one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing} 
+        new{T,CO,TR,TP,TT}(r,φ,θ,origin,rotation)
+    end
+end
+
+
+function Ellipsoid(::Type{T}=Float64; r = 1.0, φ = nothing, θ = nothing, origin = zero(CartesianPoint{T}), 
+    rotation = one(SMatrix{3, 3, T, 9}), boundary = :closed) where {T}
+
+    CO = _boundary_from_symbol(boundary)
+    return Ellipsoid{T,CO}(T.(r), φ, θ, origin, rotation)
 end
 
 Ellipsoid{T,CO,TR,TP,TT}( e::Ellipsoid{T,CO,TR,TP,TT}; COT = CO,
             origin::CartesianPoint{T} = e.origin,
             rotation::SMatrix{3,3,T,9} = e.rotation) where {T,CO<:Union{ClosedPrimitive, OpenPrimitive},TR,TP,TT} =
-    Ellipsoid{T,COT,TR,TP,TT}(e.r, e.φ, e.θ, origin, rotation)
+    Ellipsoid{T,COT,TR,TP,TT}(T; r = e.r, φ = e.φ, θ = e.θ, origin = origin, rotation = rotation)
```

Comment:
You cannot use keyword arguments here if you use the methods with curly brackets (now that `@with_kw` was removed)